csharp/src/Google.Protobuf/WellKnownTypes/AnyPartial.cs

```diff
@@ -120,6 +120,26 @@ public bool TryUnpack<T>(out T result) where T : IMessage, new()
             return true;
         }
 
+        /// <summary>
+        /// Attempts to unpack the content of this Any message into one of the message types
+        /// in the given type registry, based on the type URL.
+        /// </summary>
+        /// <param name="registry">The type registry to consult for messages.</param>
+        /// <returns>The unpacked message, or <c>null</c> if no matching message was found.</returns>
+        public IMessage UnpackMessage(TypeRegistry registry)

```

Comment:
Another overload for this could be parameter-less, and could attempt to build the TypeRegistry from all dependencies. A little heavy on reflection but a type registry built like that could be lazy and cached.

But this of course, could not be done at all, or in another PR.

Reply:
Do you mean assembly scanning to find all IMessage types?

Reply:
Yep, that's what I meant, assembly scanning to find all descriptors from which to build a TypeRegistry. As I said heavy on the reflection :).

Chatting to Jon this morning, he mentioned maybe adding a `TypeRegistry.LoadTheWorld()` and then use that `TypeRegistry` here, which would make it super explicit.

But again, either variant is just an extra feature on top of this one, so that calling code doesn't even have to know how to build the appropiate `TypeRegistry` that will contain the potential actual types of `Any`.

Reply:
Ok. We (.NET team) try to avoid assembly scanning these days. It isn't trimmer/AOT friendly. `TypeRegistry.LoadTheWorld()` is ok-ish because it isolates the assembly scanning to one API.